### PR TITLE
Support multiple SPI memory chip selects

### PR DESCRIPTION
**Pull request by alastairpatrick**

### CODE REVIEW

File: src/ssram.c [hunk 1]
```diff
@@ -3,40 +3,58 @@
...
 
-    gpio_put(ICE_SSRAM_SPI_CS_PIN, true);
-    sleep_us(1); // We use a pullup on the board so it needs a bit of time to settle
+    gpio_put(cs_pin, false);
 }
 
 /// Select the SRAM for SPI transaction.
-/// Unlike regular SPI, PSRAM CS is active high.
 static void ice_ssram_deselect(void) {
+    assert(g_cs_pin >= 0);
```
Comment (josuah) [inline src/ssram.c:45]: what about using an unsigned type here?

Comment (josuah) [reply src/ssram.c:45]: It looks like the pico-c-sdk uses `uint` internally https://raspberrypi.github.io/pico-sdk-doxygen/group__hardware__gpio.html

File: src/ssram.c [hunk 2]
```diff
@@ -92,7 +119,11 @@ void ice_ssram_init(int irq) {
     }
 }
 
+// Release hardware resources for SSRAM
 void ice_ssram_deinit(void) {
+    // Wait for previous transactions from this library to terminate
+    ice_ssram_wait();
```
Comment (josuah) [inline src/ssram.c:125]: I thought I added it back, but looks like not! Sorry about that.

**Review by josuah — APPROVED** [review]
Thank you for this improvement.
It also corrects my latest breakage from the previous edit, I might have added some of these changes back on another PR.
I will make sure that my changes do get in the way of this merge request.

File: src/ssram.c (hunk 1 above)
Comment (alastairpatrick) [reply src/ssram.c:45]: It's signed because it uses a negative value of g_cs_pin to signify that no memory chip is currently selected.

**Pull request merged**